### PR TITLE
Remove gp_fault_strategy catalog table and corresponding code.

### PR DESCRIPTION
**Pull request by ashwinstar**
Using gp_segment_configuration catalog table easily can find if mirrors exist or
not, do not need special table to communicate the same. Earlier
gp_fault_strategy used to convey 'n' for mirrorless system, 'f' for replication
and 's' for san mirrors. Since support for 's' was removed in 5.0 only purpose
gp_fault_strategy served was mirrored or not mirrored system. Hence deleting the
gp_fault_strategy table and at required places using gp_segment_configuration to
find the required info.

### CODE REVIEW

File: gpMgmt/bin/gpdbrestore [hunk 1]
```diff
@@ -337,11 +337,11 @@ class GpdbRestore(Operation):
         """
         Checking cluster status, no primary should be down.
         """
-        fault_action = self.context.gparray.getFaultStrategy()
+        fault_action = self.context.gparray.hasMirrors
```
Comment (Chibin) [inline gpMgmt/bin/gpdbrestore]: The value eventually gets used as ```        logger.info("Array fault tolerance      = %s" % info['fault_action'])``` in `function _output_info` (line 311).
if you make the change, you'll also need to change line 350.

Perhaps change the variable from `fault_action` to `has_mirrors`.

Then we can say something like ``` logger.info("Has mirrors      = %s" % info['has_mirrors'])```

Comment (Chibin) [reply gpMgmt/bin/gpdbrestore]: The change is related to gpdbrestore @khuddlefish

File: gpMgmt/bin/gpstart [hunk 2]
```diff
@@ -478,12 +478,9 @@ class GpStart:
             raise ExceptionNoStackTraceNeeded(
                 "Found a System Expansion Setup in progress. Please run 'gpexpand --rollback'")
 
-        strategy = self.gparray.getFaultStrategy()
-        logger.debug("Strategy in gparray is %s" % strategy)
+        logger.debug("gparray has mirrors %d" % self.gparray.hasMirrors)
```
[the remark this answers is not in the document]
Comment (ashwinstar) [reply gpMgmt/bin/gpstart]: Sure changed as suggested.

File: gpMgmt/bin/gpdbrestore (hunk 1 above)
Comment (ashwinstar) [reply gpMgmt/bin/gpdbrestore]: Made the change, @Chibin @khuddlefish please take a look.

File: gpMgmt/bin/gpdbrestore [hunk 3]
```diff
@@ -311,7 +311,7 @@ class GpdbRestore(Operation):
             logger.info("Restore global objects     = On")
         else:
             logger.info("Restore global objects     = Off")
-        logger.info("Array fault tolerance      = %s" % info['fault_action'])
+        logger.info("Array has mirrors      = %s" % ("Yes" if info['hasMirrors'] else "No"))
```
Comment (khuddlefish) [inline gpMgmt/bin/gpdbrestore]: It looks like the key is 'has_mirrors' not 'hasMirrors'

**Comment by ashwinstar** [comment]
On Thu, Sep 14, 2017 at 2:14 PM, Karen Huddleston <notifications@github.com>
wrote:

> *@khuddlefish* commented on this pull request.
> ------------------------------
>
> In gpMgmt/bin/gpdbrestore
> <https://github.com/greenplum-db/gpdb/pull/3265#discussion_r139012935>:
>
> > @@ -311,7 +311,7 @@ class GpdbRestore(Operation):
>              logger.info("Restore global objects     = On")
>          else:
>              logger.info("Restore global objects     = Off")
> -        logger.info("Array fault tolerance      = %s" % info['fault_action'])
> +        logger.info("Array has mirrors      = %s" % ("Yes" if info['hasMirrors'] else "No"))
>
> It looks like the key is 'has_mirrors' not 'hasMirrors'
>
Sure yes, *python*. Thank You fixed it.

**Pull request merged**